### PR TITLE
improvement(nemesis): Filter also according to default values

### PR DESCRIPTION
**Pull request by pehala**
Alternative to #11179.
This PR fixes one problem, the fact that when you filter nemesis, only the concrete class parameters are used, any parameter from the subclass (e.g. the default) wont be utilized. 

My proposed solution is to extract flags into separate class that will be taken into account when accessing the parameters as well. This has consequences of adding a lot of lines into nemesis.yml and some minor adjustments in nemesis and tests. I also added filtering to the context parameters to filter out private properties (like `__dict__` or others)

I also added tests for this and the empty `logical_phrase` case

### Testing

- [x] Unit tests

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

File: sdcm/nemesis_registry.py [hunk 1]
```diff
@@ -90,7 +93,9 @@ def gather_properties(self) -> Tuple[Dict[SourceType, Dict[str, bool]], Dict[str
         method_properties = {}
         for subclass in self.get_subclasses():
             properties_list = {}
-            for attribute in subclass.__dict__.keys():
+            keys = dict(**self.flag_class.__dict__)
```
Comment (vponomaryov) [inline sdcm/nemesis_registry.py:96]: Please, rename the `keys` dictionary to more correct name.
For example - `nemesis_params`, `nemesis_flags` or `nemesis_filters`.

File: unit_tests/nemesis/test_registry.py [hunk 2]
```diff
@@ -74,7 +79,8 @@ def registry():
         ("flag_d", {CustomNemesisD}),
         ("flag_common", {CustomNemesisA, CustomNemesisB, CustomNemesisD}),
         ("", {CustomNemesisA, CustomNemesisB, CustomNemesisC, CustomNemesisD}),
-        (None, {CustomNemesisA, CustomNemesisB, CustomNemesisC, CustomNemesisD})
+        (None, {CustomNemesisA, CustomNemesisB, CustomNemesisC, CustomNemesisD}),
+        ("flag_true", {CustomNemesisB, CustomNemesisC, CustomNemesisD}),  # flag_true is set to True in CustomNemesisA
```
Comment (vponomaryov) [inline unit_tests/nemesis/test_registry.py:83]: Typo, `flag_true` is set to `False` in the `CustomNemesisA` class.

File: unit_tests/nemesis/test_registry.py [hunk 3]
```diff
@@ -7,13 +7,17 @@
...
+class FlagClass:
     flag_a = False
     flag_b = False
     flag_c = False
     flag_d = False
     flag_common = False
+    flag_true = True
+
+
+class TestNemesis(FlagClass):
```
Comment (vponomaryov) [inline unit_tests/nemesis/test_registry.py:19]: Intuitively I would expect the `CustomNemesisD` to be renamed to the `CustomNemesisAD`

**Review by vponomaryov — COMMENTED** [review]
See inline comments, everything else LGTM
